src/driver/op.rs

```diff
@@ -149,31 +172,57 @@ where
                     Lifecycle::Completed(cqe) => {
                         driver.ops.remove(me.index);
                         me.index = usize::MAX;
-
                         Poll::Ready(me.data.take().unwrap().complete(cqe))
                     }
+                    Lifecycle::CompletionList(..) => {
+                        unreachable!("No `more` flag set for SingleCQE")
+                    }
                 }
             })
         })
     }
 }
 
-impl<T> Drop for Op<T> {
+/// The operation may have pending cqe's not yet processed.
+/// To manage this, the lifecycle associated with the Op may if required
+/// be placed in LifeCycle::Ignored state to handle cqe's which arrive after
+/// the Op has been dropped.
+impl<T, CqeType> Drop for Op<T, CqeType> {
     fn drop(&mut self) {
+        use std::mem;
+
         CONTEXT.with(|runtime_context| {
             runtime_context.with_driver_mut(|driver| {
-                let lifecycle = match driver.ops.get_mut(self.index) {
-                    Some(lifecycle) => lifecycle,
-                    None => return,
+                // Get the Op Lifecycle state from the driver
+                let (lifecycle, completions) = match driver.ops.get_mut(self.index) {
+                    Some(val) => val,
+                    None => {
+                        // Op dropped after the driver
```

Comment:
I like the addition of this comment. It highlights a potential new problem for me that I don't propose we address with this PR. Namely, we should allow a silent drop here if the driver truly has dropped before the Op, but I think we should panic if the driver exists but the slab entry for this Op's self.index is returning None, because that shows a serious invariant is being violated - if this can happen, then a cqe could come in for a slab index that had already been given to a new Op.